apps/root/src/pages/token-profile/historical-prices/index.tsx

```diff
@@ -0,0 +1,356 @@
+                    defaultMessage="Transfered:"
+                  />
+                )}
+              </Typography>
+              <Typography variant="bodySmallRegular" color={({ palette: { mode } }) => colors[mode].typography.typo3}>
+                <Address address={user} trimAddress />
+              </Typography>
+            </ContainerBox>
+            <ContainerBox alignItems="center" gap={1}>
+              <TokenIcon token={value.token} size={5} />
+              <Typography variant="bodySmallBold" color={({ palette: { mode } }) => colors[mode].typography.typo2}>
+                {formatCurrencyAmount({ amount: value.amount.amount, token: value.token, intl })}
+              </Typography>
+              <Typography variant="bodySmallRegular" color={({ palette: { mode } }) => colors[mode].typography.typo3}>
+                (${formatUsdAmount({ intl, amount: value.amount.amountInUSD })})
+              </Typography>
+            </ContainerBox>
+          </ContainerBox>
+        ))
+      )}
+    </StyledTooltipContainer>
+  );
+};
+
+const PeriodMap: Record<Exclude<GraphContainerPeriods, GraphContainerPeriods.all>, TimeString> = {
+  [GraphContainerPeriods.day]: '30m',
+  [GraphContainerPeriods.week]: '3h',
+  [GraphContainerPeriods.month]: '12h',
+  [GraphContainerPeriods.year]: '1w',
+};
+
+const DATA_POINTS = 200;
+
+const TokenHistoricalPrices = ({ token }: TokenHistoricalPricesProps) => {
+  const mode = useThemeMode();
+  const priceService = usePriceService();
+  const providerService = useProviderService();
+  const [selectedPeriod, setSelectedPeriod] = React.useState<GraphContainerPeriods>(GraphContainerPeriods.month);
+  const prevPeriod = usePrevious(selectedPeriod);
+  const [isLoadingHistorical, setIsLoadingHistorical] = React.useState(true);
+  const [graphPrices, setGraphPrices] = React.useState<Partial<Record<GraphContainerPeriods, PriceResult[]>>>({});
+
+  const handlePeriodUpdate = (newPeriod: GraphContainerPeriods) => {
+    if (!graphPrices[newPeriod]) {
+      // When period changes, the useMemo renders no data before the isLoading is set to true by the useEffect
+      // By setting it here, we avoid the flickering of the graph
+      setIsLoadingHistorical(true);
+    }
+
+    setSelectedPeriod(newPeriod);
+  };
+
+  React.useEffect(() => {
+    const handleAllTimeframe = async () => {
+      const firstTimestamp = await providerService.getBlockTimestamp(token.chainId, 1n);
+      const currentTimestamp = Math.floor(Date.now() / 1000);
+
+      const totalDataPoints = Math.floor((currentTimestamp - firstTimestamp) / (24 * 60 * 60));
+
+      if (totalDataPoints < 30) {
+        // Fresh chain
+        return {
+          period: PeriodMap[GraphContainerPeriods.day],
+          span: DATA_POINTS,
+        };
+      }
+
+      let span = totalDataPoints;
+
+      let periodDays = 1;
+      while (span > DATA_POINTS) {
+        span = Math.floor(span / 3);
+        periodDays = periodDays * 3;
```

Comment:
so if there are a lot of data points we make 1 per 3 days?

Reply:
This `while` ensures the number is between 60 and 200 aproximately. So if there are a LOT of data points, we reduce them until we are below 200
For instance, in the case of WETH, we have 3295 days (6,5 years). So we request a `period` of `27d` with a `span` of `122`.

Reply:
Do you know if the query takes longer if we have a smaller period?

Reply:
Couldn't find a strict relation between period and response time